pkg/plugins/client.go

```diff
@@ -0,0 +1,101 @@
+package plugins
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+const (
+	versionMimetype = "appplication/vnd.docker.plugins.v1+json"
+	defaultTimeOut  = 120
+)
+
+func NewClient(addr string) *Client {
+	// No TLS. Hopefully this discourages non-local plugins
```

Comment:
Ideally it would just be Unix sockets. In fact, I'd suggest limiting it to that for now because the entire system was designed for only local plugins.